config/config-api/src/main/java/com/thoughtworks/go/util/CachedDigestUtils.java

```diff
@@ -49,6 +51,21 @@ public static String sha256Hex(String string) {
         return digest;
     }
 
+    public static String md5Hex(Object serializable) {
+        return objectPools.computeDigest(DigestObjectPools.MD_5, new DigestObjectPools.DigestOperation() {
+            public String perform(MessageDigest messageDigest) {
+                try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
```

Comment:
This can possibly create a huge byte array in memory, also since the implementation is such that the array needs "resizing" once it nears capacity.

A better alternative is to use `DigestOutputStream` wrapped around a `NullOutputStream` like here:
https://github.com/gocd/gocd/blob/5c7c6e382613f4efd79fbe5ce771284296c48500/server/src/main/java/com/thoughtworks/go/server/dashboard/GoDashboardPipelineGroup.java#L51-L74

This essentially has a memory footprint that is determined by whatever is needed by the digest algorithm.